tests/Unit/MicroplateSet/MicroplateSetTest.php

```diff
@@ -0,0 +1,155 @@
+
+        $setPositionLowerThanMin = 0;
+        self::expectExceptionObject(new \OutOfRangeException("Expected a position between 1-48, got: {$setPositionLowerThanMin}"));
+        $microplateSet->locationFromPosition($setPositionLowerThanMin, FlowDirection::COLUMN());
+    }
+
+    /**
+     * @dataProvider dataProvider12Well
+     */
+    public function testSetLocationFromSetPositionFor12Wells(int $position, string $coordinateString, string $plateID): void
+    {
+        $microplateSet = new MicroplateSetABCD(new CoordinateSystem12Well());
+
+        $location = $microplateSet->locationFromPosition($position, FlowDirection::COLUMN());
+        self::assertSame($location->coordinate->toString(), $coordinateString);
+        self::assertSame($location->plateID, $plateID);
+    }
+
+    /**
+     * @return list<array{position: int, coordinateString: string, plateID: string}>
+     */
+    public static function dataProvider12Well(): array
+    {
+        return
+            [
+                [
+                    'position' => 1,
+                    'coordinateString' => 'A1',
+                    'plateID' => 'A',
+                ], [
+                    'position' => 2,
+                    'coordinateString' => 'B1',
+                    'plateID' => 'A',
+                ], [
+                    'position' => 3,
+                    'coordinateString' => 'C1',
+                    'plateID' => 'A',
+                ], [
+                    'position' => 12,
+                    'coordinateString' => 'C4',
+                    'plateID' => 'A',
+                ], [
+                    'position' => 13,
+                    'coordinateString' => 'A1',
+                    'plateID' => 'B',
+                ], [
+                    'position' => 48,
+                    'coordinateString' => 'C4',
+                    'plateID' => 'D',
+                ],
+            ];
+    }
+
+    /**
+     * @dataProvider dataProvider96Well
+     */
+    public function testSetLocationFromSetPositionFor96Wells(int $position, string $coordinateString, string $plateID): void
+    {
+        $microplateSet = new MicroplateSetABCD(new CoordinateSystem96Well());
+
+        $location = $microplateSet->locationFromPosition($position, FlowDirection::COLUMN());
+        self::assertSame($coordinateString, $location->coordinate->toString());
+        self::assertSame($plateID, $location->plateID);
+    }
+
+    /**
+     * @return list<array{position: int, coordinateString: string, plateID: string}>
+     */
+    public static function dataProvider96Well(): array
+    {
+        return
+            [
+                [
+                    'position' => 1,
+                    'coordinateString' => 'A1',
+                    'plateID' => 'A',
+                ], [
+                    'position' => 2,
+                    'coordinateString' => 'B1',
+                    'plateID' => 'A',
+                ], [
```

Comment:
```suggestion
        yield [
            'position' => 1,
            'coordinateString' => 'A1',
            'plateID' => 'A',
        ];
        yield [
            'position' => 2,
            'coordinateString' => 'B1',
            'plateID' => 'A',
        ];
```

And so on. Reduces indentation by 8. Same in the other data provider.